cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -865,19 +865,25 @@ void IndexMap::global_to_local(std::span<const std::int64_t> global,
   for (std::size_t i = 0; i < _ghosts.size(); ++i)
     global_to_local[i] = {_ghosts[i], i + local_size};
 
-  std::sort(global_to_local.begin(), global_to_local.end());
+  std::sort(global_to_local.begin(), global_to_local.end(),
+            [](auto a, auto b) { return a.first < b.first; });
   std::transform(global.begin(), global.end(), local.begin(),
                  [range = _local_range,
                   &global_to_local](std::int64_t index) -> std::int32_t
                  {
                    if (index >= range[0] and index < range[1])
                      return index - range[0];
+                   else if (index < global_to_local.front().first
```

Comment:
Is this needed? Seems like it will be caught below.

Reply:
Where is the caught below?
And even if it is caught for an element outside of the range (on the upper bound) it requires looping through the whole array?

Reply:
Removing as it doesn't work with this check in the corner case where a process has no ghosts.